simpa/utils/calculate.py

```diff
@@ -28,8 +28,12 @@ def calculate_oxygenation(molecule_list):
     elif hbO2 is None:
         hbO2 = 0
 
-    if hb + hbO2 < 1e-10:  # negative values are not allowed and division by (approx) zero
-        return None        # will lead to negative side effects.
+    if isinstance(hb, np.ndarray) or isinstance(hbO2, np.ndarray):
+        if (hb + hbO2 < 1e-10).any():  # negative values are not allowed and division by (approx) zero
+            return None                # will lead to negative side effects.
+    else:
+        if hb + hbO2 < 1e-10:
+            return None
```

Comment:
You mean additionally to setting it None?
i.e. in epidermis models or in water background, it would be expected to not get any sO2 values.
So raising an exception would not be ideal, I think...

Reply:
I mean, if the user defined Hb or HbO2 in the volume, they will expect it to be present at the end and just realizing afterwards that is None will most likely just create more confusion. If ther eis only water in the volume then this part of the code should, ideally, not be reached anyways.